src/pages/Tags/Tags.tsx

```diff
@@ -1,9 +1,242 @@
+        type: 'text',
+        rules: {
+          required: true,
+          validate: tagInputValidator,
+        },
+        className: classes.tagInputField,
+      },
+    ])
+  }
+
+  const onTagsSubmit: SubmitHandler<FormValues> = useCallback(
+    async (values) => {
+      const { tagCategorySelection, tagInput } = values
+      const transformedObject = flatMap(tagInput, (tagInputValue) =>
+        map(tagCategorySelection, (tagCategoryValue) => {
+          return {
+            type: tagCategoryValue,
+            name: tagInputValue,
+          }
+        })
+      )
+      const payload: TagsType = {
+        tags: transformedObject,
+      }
+
+      try {
+        await createTags(payload)
+        showNotification({
+          type: NotificationTypes.Success,
+          title: t('notification.announcement'),
+          content: t('success.tag_added'),
+        })
+        reset()
+      } catch (errorData) {
+        showValidationErrorMessage(errorData)
+      }
+    },
+    [createTags, reset, t]
+  )
+
+  const groupedCategoryData = groupBy(tags, 'type')
+  const uniqueCategoryTypes = Object.keys(groupedCategoryData)
+
+  const areInputFieldsDirty = !!(
+    dirtyFields.tagCategorySelection && dirtyFields.tagInput
+  )
+
+  if (isFetchingTags) {
+    return <Loader loading />
+  }
+
+  return (
+    <>
+      <div className={classes.tagsHeader}>
+        <h1>{t('tag.tag_management')}</h1>
+        <Tooltip
+          title={t('cheat_sheet.user_management.title')}
+          modalContent={<TagsCheatSheet />}
+        />
+      </div>
+      <Container className={classes.tagsContainer}>
+        <div>
+          <h4 className={classes.addingTag}>{t('tag.adding_tag')}</h4>
+          <p>{t('tag.naming_tag')}</p>
+        </div>
+
+        <div className={classes.addingTagsSeparator} />
+
+        <div className={classes.tagsSection}>
+          <DynamicForm fields={inputFields} control={control} />
+          <Button
+            appearance={AppearanceTypes.Text}
+            iconPositioning={IconPositioningTypes.Left}
+            icon={Add}
+            className={classes.addNewRow}
+            children={t('tag.add_new_row')}
+            onClick={addInputField}
+            hidden={!includes(userPrivileges, Privileges.AddTag)}
+          />
+        </div>
+
+        <div className={classes.categorySection}>
+          <DynamicForm fields={categoryFields} control={control} />
+          <Button
+            children={t('button.add')}
+            type="submit"
+            ariaLabel={t('button.add')}
+            className={classes.addButton}
+            onClick={handleSubmit(onTagsSubmit)}
+            loading={isCreatingTags}
+            disabled={!areInputFieldsDirty}
+            hidden={!includes(userPrivileges, Privileges.AddTag)}
+          />
+        </div>
+      </Container>
+
+      <div className={classes.categoryContainer}>
+        {map(uniqueCategoryTypes, (type) => (
```

Comment:
**[non-blocker]**

There actually is no need to split uniqueCategoryTypes from groupedCategoryData for mapping this data.
you could instead use 

map(groupedCategoryData, (tags, type) => ...

instead and use the "type" the same way it is used now + replace groupedCategoryData[type] with "tags". 

Not a major issue, just would mean less code and less complexity